shaptools/hdb_connector/connectors/pyhdb_connector.py

```diff
@@ -53,18 +53,20 @@ def connect(self, host, port=30015, **kwargs):
 
     def query(self, sql_statement):
         """
-        Query a sql statement and return response
+        Query a sql statement and return a response and a response meta data
         """
         self._logger.info('executing sql query: %s' % sql_statement)
         try:
             cursor = self._connection.cursor()
             cursor.execute(sql_statement)
             result = cursor.fetchall()
+            meta_data = cursor.description
             cursor.close()
         except pyhdb.exceptions.DatabaseError as err:
             raise base_connector.QueryError('query failed: {}'.format(err))
         self._logger.info('query result: %s' % result)
-        return result
+        self._logger.info('query result meta data: %s' % meta_data)
+        return meta_data, result
```

Comment:
The main purpose of this method is to return the resultset, so it should come always in the first position.  Also, would be better to have an object where we could improve over time, including attributes and methods. This avoids complexity on the caller code.